back_pressure.go

```diff
@@ -0,0 +1,72 @@
+package remotedialer
+
+import (
+	"sync"
+)
+
+type backPressure struct {
+	cond   sync.Cond
+	c      *connection
+	paused bool
+}
+
+func newBackPressure(c *connection) *backPressure {
+	return &backPressure{
+		cond: sync.Cond{
+			L: &sync.Mutex{},
+		},
+		c:      c,
+		paused: false,
+	}
+}
+
+func (b *backPressure) OnPause() {
+	b.cond.L.Lock()
+	defer b.cond.L.Unlock()
+
+	b.paused = true
+	b.cond.Broadcast()
+}
+
+func (b *backPressure) OnResume() {
+	b.cond.L.Lock()
+	defer b.cond.L.Unlock()
+
+	b.paused = false
+	b.cond.Broadcast()
```

Comment:
Need an advanced sync lecture for this... :astonished:

Reply:
how do you mean?